sqs/src/main/scala/akka/stream/alpakka/sqs/scaladsl/SqsSource.scala

```diff
@@ -43,15 +44,24 @@ object SqsSource {
           case Some(t) => requestBuilder.visibilityTimeout(t.toSeconds.toInt).build()
         }
       }
-      .via(
-        BalancingMapAsyncUnordered[ReceiveMessageRequest, ReceiveMessageResponse](
-          settings.parallelRequests,
-          sqsClient.receiveMessage(_).toScala,
-          response => if (response.messages().isEmpty) 1 else settings.parallelRequests
-        )
-      )
+      .via(resolveHandler(settings.parallelRequests))
       .map(_.messages().asScala.toList)
       .takeWhile(messages => !settings.closeOnEmptyReceive || messages.nonEmpty)
       .mapConcat(identity)
       .buffer(settings.maxBufferSize, OverflowStrategy.backpressure)
+
+  private def resolveHandler(parallelism: Int)(implicit sqsClient: SqsAsyncClient) = {
+    if (parallelism == 1) {
+      MapAsyncUnordered[ReceiveMessageRequest, ReceiveMessageResponse](
+        parallelism,
+        sqsClient.receiveMessage(_).toScala
+      )
```

Comment:
You can replace this with `Flow[ReceiveMessageRequest].mapAsyncUnordered(parallelism)(sqsClient.receiveMessage(_).toScala)` and then you will not need to import `MapAsyncUnordered` from the `impl` package.